src/Http/Controllers/ExcelController.php

```diff
@@ -17,19 +17,30 @@ class ExcelController extends Controller
      * @param Request         $request
      * @param ResponseFactory $response
      *
-     * @return BinaryFileResponse
+     * @return Response
      * @throws ValidationException
      */
-    public function download(Request $request, ResponseFactory $response): BinaryFileResponse
+    public function download(Request $request, ResponseFactory $response): Response
     {
         $data = $this->validate($request, [
             'path'     => 'required',
             'filename' => 'required',
         ]);
 
-        return $response->download(
-            decrypt($data['path']),
-            $data['filename']
-        )->deleteFileAfterSend($shouldDelete = true);
+        $decryptedPath = decrypt($data['path']);
+
+        if (config('excel.temporary_files.remote_disk')) {
+            app()->terminating(function () use ($decryptedPath) {
+                \Storage::disk(config('excel.temporary_files.remote_disk'))->delete($decryptedPath);
```

Comment:
Can you use the FQN of Storage facade as this will break if someone decides to remove the aliases in their config.